platforms/ide/tooling-api/src/testFixtures/groovy/org/gradle/integtests/tooling/fixture/ToolingApiDistributionResolver.groovy

```diff
@@ -16,102 +16,93 @@
-            this.resolutionServices.resolveRepositoryHandler.maven { url = localRepository.toURI() }
-        }
-    }
-
-    ToolingApiDistributionResolver withRepository(String repositoryUrl) {
-        resolutionServices.resolveRepositoryHandler.maven { url = repositoryUrl }
-        this
-    }
-
-    ToolingApiDistributionResolver withDefaultRepository() {
-        withRepository(RepoScriptBlockUtil.gradleRepositoryMirrorUrl())
-    }
-
-    ToolingApiDistributionResolver withExternalToolingApiDistribution() {
-        this.useExternalToolingApiDistribution = true
-        this
-    }
 
     ToolingApiDistribution resolve(String toolingApiVersion) {
         if (!distributions[toolingApiVersion]) {
             if (useToolingApiFromTestClasspath(toolingApiVersion)) {
                 distributions[toolingApiVersion] = new TestClasspathToolingApiDistribution()
             } else if (CommitDistribution.isCommitDistribution(toolingApiVersion)) {
                 File toolingApiJar = CommitDistribution.getToolingApiJar(toolingApiVersion)
-                List<File> slf4j = resolveDependency("org.slf4j:slf4j-api:1.7.25").toList()
-                distributions[toolingApiVersion] = new ExternalToolingApiDistribution(toolingApiVersion, slf4j + toolingApiJar)
+                File slf4jApi = locateLocalSlf4j()
+                distributions[toolingApiVersion] = new ExternalToolingApiDistribution(toolingApiVersion, [slf4jApi, toolingApiJar])
             } else {
-                distributions[toolingApiVersion] = new ExternalToolingApiDistribution(toolingApiVersion, resolveDependency("org.gradle:gradle-tooling-api:$toolingApiVersion"))
+                File toolingApiJar = locateToolingApi(toolingApiVersion)
+                File slf4jApi = locateLocalSlf4j()
+                distributions[toolingApiVersion] = new ExternalToolingApiDistribution(toolingApiVersion, [slf4jApi, toolingApiJar])
             }
         }
         distributions[toolingApiVersion]
     }
 
-    private Collection<File> resolveDependency(String dependency) {
+    private File locateToolingApi(String version) {
+        def relativePath = "org/gradle/gradle-tooling-api/$version/gradle-tooling-api-${version}.jar"
+
+        File localRepository = buildContext.localRepository
+        if (localRepository) {
+            Path jarFile = localRepository.toPath().resolve(relativePath)
+            if (Files.exists(jarFile)) {
+                return jarFile.toFile()
+            }
+        }
+
+        File destination = buildContext.tmpDir.file("gradle-tooling-api-${version}.jar")
+        if (!destination.exists()) {
+            withRetries {
+                def url = RepoScriptBlockUtil.gradleRepositoryMirrorUrl() + "/" + relativePath
+                LOGGER.warn("Downloading tooling API {}", version)
+                try (InputStream stream = new URL(url).openStream()) {
+                    Files.copy(stream, destination.toPath(), StandardCopyOption.REPLACE_EXISTING)
```

Comment:
`StandardCopyOption.REPLACE_EXISTING` probably not necessary, as we already check it doesn't exist